dist/js/select2.full.js

```diff
@@ -1386,9 +1386,14 @@ S2.define('select2/selection/base',[
     var id = container.id + '-container';
     var resultsId = container.id + '-results';
     var searchHidden = this.options.get('minimumResultsForSearch') === Infinity;
+    var isRequired = this.options.get('required') === true;
 
     this.container = container;
 
+    if (isRequired) {
```

Comment:
We have a PR under review that drops select2

https://github.com/woocommerce/woocommerce/pull/48731

So these changes might not be required.